File: app/buildz/tmp/kapt3/stubs/debug/com/bits/bee/bpmc/presentation/ui/pos/pos_item/ItemPosAdapter.java
package com.bits.bee.bpmc.presentation.ui.pos.pos_item;

import java.lang.System;

/**
 * Created by aldi on 22/04/22.
 */
@kotlin.Metadata(mv = {1, 7, 1}, k = 1, d1 = {"\u0000J\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\u0010\u0002\n\u0002\b\u0002\n\u0002\u0010 \n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0010\u000e\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0002\b\u0004\n\u0002\u0010\b\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0002\b\u0006\u0018\u00002\u000e\u0012\u0004\u0012\u00020\u0002\u0012\u0004\u0012\u00020\u00030\u0001:\u0002\u001c\u001dB;\u0012\u0012\u0010\u0004\u001a\u000e\u0012\u0004\u0012\u00020\u0002\u0012\u0004\u0012\u00020\u00060\u0005\u0012\u0012\u0010\u0007\u001a\u000e\u0012\u0004\u0012\u00020\u0002\u0012\u0004\u0012\u00020\u00060\u0005\u0012\f\u0010\b\u001a\b\u0012\u0004\u0012\u00020\n0\t\u00a2\u0006\u0002\u0010\u000bJ\u0010\u0010\f\u001a\u00020\r2\u0006\u0010\u000e\u001a\u00020\rH\u0002J\u0010\u0010\u000f\u001a\u00020\u00102\u0006\u0010\u0011\u001a\u00020\u0002H\u0002J\u0018\u0010\u0012\u001a\u00020\u00062\u0006\u0010\u0013\u001a\u00020\u00032\u0006\u0010\u0014\u001a\u00020\u0015H\u0016J\u0018\u0010\u0016\u001a\u00020\u00032\u0006\u0010\u0017\u001a\u00020\u00182\u0006\u0010\u0019\u001a\u00020\u0015H\u0016J\u0014\u0010\u001a\u001a\u00020\u00062\f\u0010\u001b\u001a\b\u0012\u0004\u0012\u00020\n0\tR\u001a\u0010\u0004\u001a\u000e\u0012\u0004\u0012\u00020\u0002\u0012\u0004\u0012\u00020\u00060\u0005X\u0082\u0004\u00a2\u0006\u0002\n\u0000R\u001a\u0010\u0007\u001a\u000e\u0012\u0004\u0012\u00020\u0002\u0012\u0004\u0012\u00020\u00060\u0005X\u0082\u0004\u00a2\u0006\u0002\n\u0000R\u0014\u0010\b\u001a\b\u0012\u0004\u0012\u00020\n0\tX\u0082\u000e\u00a2\u0006\u0002\n\u0000\u00a8\u0006\u001e"}, d2 = {"Lcom/bits/bee/bpmc/presentation/ui/pos/pos_item/ItemPosAdapter;", "Landroidx/paging/PagingDataAdapter;", "Lcom/bits/bee/bpmc/domain/model/Item;", "Landroidx/recyclerview/widget/RecyclerView$ViewHolder;", "onItemClicK", "Lkotlin/Function1;", "", "onMinusClick", "saledList", "", "Lcom/bits/bee/bpmc/domain/model/Saled;", "(Lkotlin/jvm/functions/Function1;Lkotlin/jvm/functions/Function1;Ljava/util/List;)V", "getInitial", "", "title", "getSumQty", "Ljava/math/BigDecimal;", "item", "onBindViewHolder", "holder", "position", "", "onCreateViewHolder", "parent", "Landroid/view/ViewGroup;", "viewType", "submitSaledList", "list", "DiffCallback", "ViewHolder", "app_debug"})
public final class ItemPosAdapter extends androidx.paging.PagingDataAdapter<com.bits.bee.bpmc.domain.model.Item, androidx.recyclerview.widget.RecyclerView.ViewHolder> {
    private final kotlin.jvm.functions.Function1<com.bits.bee.bpmc.domain.model.Item, kotlin.Unit> onItemClicK = null;
    private final kotlin.jvm.functions.Function1<com.bits.bee.bpmc.domain.model.Item, kotlin.Unit> onMinusClick = null;
    private java.util.List<com.bits.bee.bpmc.domain.model.Saled> saledList;
    
    public ItemPosAdapter(@org.jetbrains.annotations.NotNull()
    kotlin.jvm.functions.Function1<? super com.bits.bee.bpmc.domain.model.Item, kotlin.Unit> onItemClicK, @org.jetbrains.annotations.NotNull()
    kotlin.jvm.functions.Function1<? super com.bits.bee.bpmc.domain.model.Item, kotlin.Unit> onMinusClick, @org.jetbrains.annotations.NotNull()
    java.util.List<com.bits.bee.bpmc.domain.model.Saled> saledList) {
        super(null, null, null);
    }
    
    public final void submitSaledList(@org.jetbrains.annotations.NotNull()
    java.util.List<com.bits.bee.bpmc.domain.model.Saled> list) {
    }
    
    @org.jetbrains.annotations.NotNull()
    @java.lang.Override()
    public androidx.recyclerview.widget.RecyclerView.ViewHolder onCreateViewHolder(@org.jetbrains.annotations.NotNull()
    android.view.ViewGroup parent, int viewType) {
        return null;
    }
    
    @java.lang.Override()
    public void onBindViewHolder(@org.jetbrains.annotations.NotNull()
    androidx.recyclerview.widget.RecyclerView.ViewHolder holder, int position) {
    }
    
    private final java.math.BigDecimal getSumQty(com.bits.bee.bpmc.domain.model.Item item) {
        return null;
    }
    
    private final java.lang.String getInitial(java.lang.String title) {
        return null;
    }
    
    @kotlin.Metadata(mv = {1, 7, 1}, k = 1, d1 = {"\u0000\u001e\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0010\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\b\u0086\u0004\u0018\u00002\u00020\u0001B\r\u0012\u0006\u0010\u0002\u001a\u00020\u0003\u00a2\u0006\u0002\u0010\u0004J\u000e\u0010\u0005\u001a\u00020\u00062\u0006\u0010\u0007\u001a\u00020\bR\u000e\u0010\u0002\u001a\u00020\u0003X\u0082\u0004\u00a2\u0006\u0002\n\u0000\u00a8\u0006\t"}, d2 = {"Lcom/bits/bee/bpmc/presentation/ui/pos/pos_item/ItemPosAdapter$ViewHolder;", "Landroidx/recyclerview/widget/RecyclerView$ViewHolder;", "binding", "Lcom/bits/bee/bpmc/databinding/ItemPosMenuBinding;", "(Lcom/bits/bee/bpmc/presentation/ui/pos/pos_item/ItemPosAdapter;Lcom/bits/bee/bpmc/databinding/ItemPosMenuBinding;)V", "bind", "", "item", "Lcom/bits/bee/bpmc/domain/model/Item;", "app_debug"})
    public final class ViewHolder extends androidx.recyclerview.widget.RecyclerView.ViewHolder {
        private final com.bits.bee.bpmc.databinding.ItemPosMenuBinding binding = null;
        
        public ViewHolder(@org.jetbrains.annotations.NotNull()
        com.bits.bee.bpmc.databinding.ItemPosMenuBinding binding) {
            super(null);
        }
        
        public final void bind(@org.jetbrains.annotations.NotNull()
        com.bits.bee.bpmc.domain.model.Item item) {
        }
    }
    
    @kotlin.Metadata(mv = {1, 7, 1}, k = 1, d1 = {"\u0000\u0018\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0010\u000b\n\u0002\b\u0004\u0018\u00002\b\u0012\u0004\u0012\u00020\u00020\u0001B\u0005\u00a2\u0006\u0002\u0010\u0003J\u0018\u0010\u0004\u001a\u00020\u00052\u0006\u0010\u0006\u001a\u00020\u00022\u0006\u0010\u0007\u001a\u00020\u0002H\u0016J\u0018\u0010\b\u001a\u00020\u00052\u0006\u0010\u0006\u001a\u00020\u00022\u0006\u0010\u0007\u001a\u00020\u0002H\u0016\u00a8\u0006\t"}, d2 = {"Lcom/bits/bee/bpmc/presentation/ui/pos/pos_item/ItemPosAdapter$DiffCallback;", "Landroidx/recyclerview/widget/DiffUtil$ItemCallback;", "Lcom/bits/bee/bpmc/domain/model/Item;", "()V", "areContentsTheSame", "", "oldItem", "newItem", "areItemsTheSame", "app_debug"})
    public static final class DiffCallback extends androidx.recyclerview.widget.DiffUtil.ItemCallback<com.bits.bee.bpmc.domain.model.Item> {
        
        public DiffCallback() {
            super();
        }
        
        @java.lang.Override()
        public boolean areItemsTheSame(@org.jetbrains.annotations.NotNull()
        com.bits.bee.bpmc.domain.model.Item oldItem, @org.jetbrains.annotations.NotNull()
        com.bits.bee.bpmc.domain.model.Item newItem) {
            return false;
        }
        
        @java.lang.Override()
        public boolean areContentsTheSame(@org.jetbrains.annotations.NotNull()
        com.bits.bee.bpmc.domain.model.Item oldItem, @org.jetbrains.annotations.NotNull()
        com.bits.bee.bpmc.domain.model.Item newItem) {
            return false;
        }
    }
}

File: app/src/main/java/com/bits/bee/bpmc/presentation/ui/pos/pos_item/ItemPosAdapter.kt
package com.bits.bee.bpmc.presentation.ui.pos.pos_item

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.core.content.ContextCompat
import androidx.core.view.isVisible
import androidx.paging.PagingDataAdapter
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.RecyclerView
import com.bits.bee.bpmc.R
import com.bits.bee.bpmc.databinding.ItemPosMenuBinding
import com.bits.bee.bpmc.domain.model.Item
import com.bits.bee.bpmc.domain.model.Saled
import com.bits.bee.bpmc.utils.BPMConstants
import com.bits.bee.bpmc.utils.CurrencyUtils
import com.bits.bee.bpmc.utils.ImageUtils
import com.bits.bee.bpmc.utils.gone
import com.bits.bee.bpmc.utils.visible
import com.squareup.picasso.Picasso
import java.io.File
import java.math.BigDecimal

/**
 * Created by aldi on 22/04/22.
 */
open class ItemPosAdapter constructor(
    private val onItemClicK : (Item) -> Unit,
    private val onMinusClick : (Item) -> Unit,
    private var saledList : List<Saled>,
    private val ukuranFont : String = "",
    private val isMultiline : Boolean = false,
    private val isMuatGambar : Boolean = false,
    private val orientation : String = ""
) : PagingDataAdapter<Item, RecyclerView.ViewHolder>(DiffCallback()) {

    fun submitSaledList(list : List<Saled>) {
        this.saledList = list
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        return ViewHolder(
            ItemPosMenuBinding.inflate(inflater, parent, false)
        )
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        val viewHolder = holder as ViewHolder
        getItem(position)?.let {
            viewHolder.bind(it)
        }
    }

    open inner class ViewHolder(private val binding : ItemPosMenuBinding) : RecyclerView.ViewHolder(binding.root){

        fun bind(item : Item){
            binding.apply {
                val qty = getSumQty(item)
                tvNamaItem.text =  item.name1

                tvNamaItem.apply {
                    maxLines = if(!isMultiline) 1 else 2
                    textSize = when(ukuranFont){
                        "Large" -> 16
                        "Small" -> 12
                        else -> 14
                    }.toFloat()
                }

                tvHarga.text = binding.root.context.getString(R.string.mata_uang_nominal,item.crcSymbol, CurrencyUtils.formatCurrency(item.price))

                tvHarga.isVisible = !item.isVariant

                tvQty?.let {
                    it.text = CurrencyUtils.formatCurrency(qty)

                    if(qty > BigDecimal.ZERO){
                        cdContent.setCardBackgroundColor(ContextCompat.getColor(root.context, R.color.red))
                        cardView.gone()
                        llMinus!!.visible()
                        llQty!!.visible()
                        tvNamaItem.setTextColor(ContextCompat.getColor(root.context, R.color.white))
                        tvHarga.setTextColor(ContextCompat.getColor(root.context, R.color.white))
                    } else {
                        cdContent.setCardBackgroundColor(ContextCompat.getColor(root.context, R.color.white))
                        cardView.visible()
                        llMinus!!.gone()
                        llQty!!.gone()
                        tvNamaItem.setTextColor(ContextCompat.getColor(root.context, R.color.black))
                        tvHarga.setTextColor(ContextCompat.getColor(root.context, R.color.black))
                    }

                    llMinus.setOnClickListener {
//                    item.qty = item.qty.subtract(BigDecimal.ZERO)
                        onMinusClick(item)
                        notifyItemChanged(absoluteAdapterPosition)
                    }
                }

                tvInisial?.apply {
                    text = ImageUtils.getInitial(item.name1)
                    textSize = ImageUtils.getFontSize(item.name1.length).toFloat()
                }

                if(!isMuatGambar) {
                    imageItem.setImageDrawable(
                        ImageUtils.generateFromInitial(
                            binding.root.context,
                            item.name1
                        )
                    )

                    llTransparent?.isVisible = false

                    tvNamaItem.setTextColor(ContextCompat.getColor(root.context, R.color.black))
                    tvHarga.setTextColor(ContextCompat.getColor(root.context, R.color.black))
                } else {
                    if(!item.tempUrl.isNullOrEmpty()) {
                        Picasso.get().load(File(BPMConstants.getDatapath() + BPMConstants.BPM_PRODUKPATH + "/", item.tempUrl!!)).into(imageItem)
                        if(orientation == BPMConstants.SCREEN_LANDSCAPE) {
                            llTransparent?.isVisible = true

                            tvNamaItem.setTextColor(
                                ContextCompat.getColor(
                                    root.context,
                                    R.color.white
                                )
                            )
                            tvHarga.setTextColor(
                                ContextCompat.getColor(
                                    root.context,
                                    R.color.white
                                )
                            )
                        }
                    } else {
                        imageItem.setImageDrawable(
                            ImageUtils.generateFromInitial(
                                binding.root.context,
                                item.name1
                            )
                        )
                        if(orientation == BPMConstants.SCREEN_LANDSCAPE) {
                            llTransparent?.isVisible = false

                            tvNamaItem.setTextColor(
                                ContextCompat.getColor(
                                    root.context,
                                    R.color.black
                                )
                            )
                            tvHarga.setTextColor(
                                ContextCompat.getColor(
                                    root.context,
                                    R.color.black
                                )
                            )
                        }
                    }
                }

                cdContent.setOnClickListener {
//                    qty = item.qty.add(BigDecimal.ZERO)
                    onItemClicK(item)
                    notifyItemChanged(absoluteAdapterPosition)
                }
            }
        }
    }

    internal fun getSumQty(item : Item) : BigDecimal {
        var qty = BigDecimal.ZERO
        saledList
            .filter { !it.isBonus }
            .forEach {
                if(item.id == it.itemId || (item.isVariant && item.itemVariantList.contains(it.itemId)))
                    qty = qty.add(it.qty)
            }
        return qty
    }

    class DiffCallback : DiffUtil.ItemCallback<Item>(){
        override fun areItemsTheSame(oldItem: Item, newItem: Item): Boolean {
            return oldItem == newItem
        }

        override fun areContentsTheSame(oldItem: Item, newItem: Item): Boolean {
            return oldItem.id == newItem.id
        }
    }
}